src/org/ods/orchestration/util/DocumentHistory.groovy

```diff
@@ -0,0 +1,332 @@
+package org.ods.orchestration.util
+
+import com.cloudbees.groovy.cps.NonCPS
+import org.ods.orchestration.service.leva.ProjectDataBitbucketRepository
+import org.ods.orchestration.util.Project.JiraDataItem
+import org.ods.util.ILogger
+import org.ods.util.IPipelineSteps
+
+/**
+ * This class contains the logic for keeping a consistent document version.
+ */
+class DocumentHistory {
+
+    static final String ADD = 'add'
+    static final String ADDED = ADD + 'ed'
+    static final String DELETE = 'discontinue'
+    static final String DELETED = DELETE + 'd'
+    static final String CHANGE = 'change'
+    static final String CHANGED = CHANGE + 'd'
+
```

Comment:
to improve encapsulation the following internal class members could be set to private